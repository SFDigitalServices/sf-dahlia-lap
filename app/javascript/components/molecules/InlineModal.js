import React from 'react'

import classNames from 'classnames'
import PropTypes from 'prop-types'

const InlineModal = ({ whiteBackground = false, children, marginBottom = false, id }) => {
  const classes = [
    'inline-modal',
    'padding-left--2x',
    'padding-right--2x',
    { 'white-background': whiteBackground },
    { 'margin-bottom--3halves': marginBottom }
  ]

  return (
    <div className={classNames(classes)} id={id}>
      {children}
    </div>
  )
}

InlineModal.propTypes = {
  whiteBackground: PropTypes.bool,
  marginBottom: PropTypes.bool,
  id: PropTypes.string
}

export default InlineModal
