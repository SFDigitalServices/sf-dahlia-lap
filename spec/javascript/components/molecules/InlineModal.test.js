import React from 'react'

import { shallow } from 'enzyme'

import InlineModal from 'components/molecules/InlineModal'

describe('InlineModal', () => {
  test('should use the inline-modal class for background if background is not specified', () => {
    const wrapper = shallow(<InlineModal />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should pass the white background prop if specified', () => {
    const wrapper = shallow(<InlineModal whiteBackground />)
    expect(wrapper).toMatchSnapshot()
  })
})
