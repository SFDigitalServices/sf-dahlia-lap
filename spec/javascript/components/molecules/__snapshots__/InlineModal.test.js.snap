// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineModal should pass the white background prop if specified 1`] = `
<div
  className="inline-modal padding-left--2x padding-right--2x white-background"
/>
`;

exports[`InlineModal should use the inline-modal class for background if background is not specified 1`] = `
<div
  className="inline-modal padding-left--2x padding-right--2x"
/>
`;
